File: server_integration_tests/servers/java_server/src/main/java/dev/graphlink/javaserver/impl/ArticleSchemaMappingsServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.ArticleSchemaMappingsService;
import dev.graphlink.javaserver.generated.services.ArticleWithCountSchemaMappingsService;
import dev.graphlink.javaserver.generated.types.Article;
import dev.graphlink.javaserver.generated.types.Author;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class ArticleSchemaMappingsServiceImpl
        implements ArticleSchemaMappingsService, ArticleWithCountSchemaMappingsService {

    @Override
    public Author articleAuthor(Article value) {
        return Data.authors.stream()
            .filter(a -> a.getId().equals(value.getAuthorId()))
            .findFirst()
            .orElse(null);
    }

    @Override
    public List<Author> articleAuthorList(Article value) {
        return Data.authors.stream()
            .filter(a -> a.getId().equals(value.getAuthorId()))
            .collect(Collectors.toList());
    }

    @Override
    public int articleWithCountCount(Article value) {
        return Data.articles.indexOf(value) + 1;
    }
}

File: server_integration_tests/kotlin/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/ArticleSchemaMappingsServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.services.ArticleSchemaMappingsService
import dev.graphlink.kotlinserverblocking.generated.services.ArticleWithCountSchemaMappingsService
import dev.graphlink.kotlinserverblocking.generated.types.Article
import dev.graphlink.kotlinserverblocking.generated.types.Author
import org.springframework.stereotype.Service

@Service
class ArticleSchemaMappingsServiceImpl :
    ArticleSchemaMappingsService, ArticleWithCountSchemaMappingsService {

    override fun articleAuthor(value: Article): Author =
        Data.authors.first { it.id == value.authorId }

    override fun articleAuthorList(value: Article): List<Author>? =
        Data.authors.filter { it.id == value.authorId }

    override fun articleWithCountCount(value: Article): Int = Data.articles.indexOf(value) + 1
}
